### PR TITLE
Miscellaneous changes

### PR DESCRIPTION
**Pull request by Vauff**
- Fixed extend time being applied twice
- Fixed extend votes being able to be rapidly chained when more than one is available
- Allowed Discord webhooks to not override default settings
- Added player IP tracking
- Deduplicated JSON parsing code
- Switched to safer method of string storage in HTTPHeader & TrackedRequest classes
- Added missing header files to VS project files
- Added missing map vote cvars to cs2fixes.cfg
- Made config file formatting consistent
- Some code cleanup in c_http debug command
- Fixed memdbgon.h not being last include in events.cpp
- Fixed !setnextmap not having correct admin flag
- Switched to simpler CON_COMMAND_CHAT macro for most map vote commands

### CODE REVIEW

**Review by xen-000 — APPROVED** [review]

File: src/playermanager.cpp [hunk 1]
```diff
@@ -138,6 +138,20 @@ bool CPlayerManager::OnClientConnected(CPlayerSlot slot, uint64 xuid)
 	ZEPlayer *pPlayer = new ZEPlayer(slot);
 	pPlayer->SetUnauthenticatedSteamId(new CSteamID(xuid));
 
+	std::string ip(pszNetworkID);
+
+	// Remove port
+	for (int i = 0; i < ip.length(); i++)
+	{
+		if (ip[i] == ':')
```
Comment (GXTX) [inline src/playermanager.cpp:146]: Consider using `ip.find()` instead.

File: src/playermanager.cpp [hunk 2]
```diff
@@ -129,7 +129,7 @@ void CPlayerManager::OnBotConnected(CPlayerSlot slot)
 	m_vecPlayers[slot.Get()] = new ZEPlayer(slot, true);
 }
 
-bool CPlayerManager::OnClientConnected(CPlayerSlot slot, uint64 xuid)
+bool CPlayerManager::OnClientConnected(CPlayerSlot slot, uint64 xuid, const char* pszNetworkID)
```
Comment (GXTX) [inline src/playermanager.cpp:132]: Should be a reference instead of a ptr

File: src/playermanager.h [hunk 3]
```diff
@@ -127,6 +129,7 @@ class ZEPlayer
 	float m_flExtendVoteTime;
 	int m_iFloodTokens;
 	float m_flLastTalkTime;
+	std::string m_strIp;
```
Comment (GXTX) [inline src/playermanager.h:132]: Consider default initializing member variables in the future

File: src/votemanager.cpp [hunk 4]
```diff
@@ -380,8 +380,8 @@ CON_COMMAND_CHAT(ve, "Vote to extend the current map.")
 			// there's an extend left after a successfull extend vote
 			g_ExtendState = EExtendState::POST_EXTEND_COOLDOWN;
 
-			// Allow another extend vote after 2 minutes
-			new CTimer(120.0f, false, []()
+			// Allow another extend vote after added time lapses
+			new CTimer(g_iExtendTimeToAdd * 60.0f, false, []()
```
Comment (GXTX) [inline src/votemanager.cpp:384]: C++std is set to C++ 20, should be using smart pointers... Same with the `delete` [here](https://github.com/Source2ZE/CS2Fixes/pull/146/files#diff-2b69d38a4a741a5b1647a74b640f5c033318011b1ab1526ab94d84883a0318a4R96)

**Pull request merged**